pkg/table/tables/tables.go

```diff
@@ -1478,11 +1478,43 @@ func (t *TableCommon) removeRowIndices(ctx table.MutateContext, h kv.Handle, rec
 	if err != nil {
 		return err
 	}
-	for _, v := range t.deletableIndices() {
+	for _, v := range t.DeletableIndices() {
 		if v.Meta().Primary && (t.Meta().IsCommonHandle || t.Meta().PKIsHandle) {
 			continue
 		}
-		vals, err := v.FetchValues(rec, nil)
+		vals, err := v.FetchValues(ctx, rec, nil)
+		if err != nil {
+			logutil.BgLogger().Info("remove row index failed", zap.Any("index", v.Meta()), zap.Uint64("txnStartTS", txn.StartTS()), zap.String("handle", h.String()), zap.Any("record", rec), zap.Error(err))
+			return err
+		}
+		if err = v.Delete(ctx, txn, vals, h); err != nil {
+			if v.Meta().State != model.StatePublic && kv.ErrNotExist.Equal(err) {
+				// If the index is not in public state, we may have not created the index,
+				// or already deleted the index, so skip ErrNotExist error.
+				logutil.BgLogger().Debug("row index not exists", zap.Any("index", v.Meta()), zap.Uint64("txnStartTS", txn.StartTS()), zap.String("handle", h.String()))
+				continue
```

Comment:
Perhaps need review by the DDL owner, why the `kv.ErrNotExist` would be returned from the deletable index? Not quite sure about it.